src/acreplace.rs

```diff
@@ -25,14 +67,33 @@ byond_fn! { setup_acreplace(key, patternsjson, replacementsjson) {
     Some("")
 } }
 
+byond_fn! { setup_acreplace_with_options(key, options_json, patterns_json, replacements_json) {
+    let options: AhoCorasickOptions = serde_json::from_str(&options_json).ok()?;
+    let patterns: Vec<String> = serde_json::from_str(&patterns_json).ok()?;
+    let replacements: Vec<String> = serde_json::from_str(&replacements_json).ok()?;
+    let ac = options.auto_configure_and_build(&patterns);
+    CREPLACE_MAP.with(|cell| {
+        let mut map = cell.borrow_mut();
+        map.insert(key.to_owned(), Replacements { automaton: ac, replacements: replacements });
+    });
+    Some("")
+} }
+
 
 byond_fn! { acreplace(key, text) {
     CREPLACE_MAP.with(|cell| -> Option<String> {
         let map = cell.borrow_mut();
-        match map.get(&key.to_owned()) {
-            Some(replacements) => Some(replacements.automaton.replace_all(text, &replacements.replacements)),
-            None => None
-        }
+        let replacements = map.get(&key.to_owned())?;
```

Comment:
Why `to_owned()` then immediately referencing it?

Reply:
autopiloted to `.to_owned()` at some point in development and then made it work with the ampersand. removed.